aws-s3-transfer-manager/src/io/path_body.rs

```diff
@@ -99,7 +99,7 @@ impl PathBodyBuilder {
                 let metadata = self.metadata.unwrap_or(fs::metadata(path.clone())?);
                 let file_size = metadata.len();
 
-                if offset >= file_size {
+                if offset > file_size {
```

Comment:
`OffsetGreaterThanFileSize` says `"offset must be less than or equal to file size but was greater than"`, but it was raised when the offset equals the file size, which seems to be a bug.

And if the offset equals to the file size, the path body will result in a 0 as the length, which seems to be expected